hyperopt/suggest.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from future import standard_library
+from builtins import object
+import logging
+import os
+
+import numpy as np
+
+from .utils import coarse_utcnow
+from . import base
+from . import tpe
+from . import hp
+
+from pandas import DataFrame
+from datetime import datetime
+
+
+class SuggestObject(object):
+    """Object for conducting search experiments.
+    """
+
+    def __init__(self, algo, domain, trials, rstate):
+        self.algo = algo
+        self.domain = domain
+        self.trials = trials
+        self.rstate = rstate
+
+    def serial_evaluate(self):
+        for trial in self.trials._dynamic_trials:
+            if trial['state'] == base.JOB_STATE_NEW:
+                result = dict(loss=None, status='ok')
+                now = coarse_utcnow()
+                trial['book_time'] = now
+                trial['refresh_time'] = now
+                trial['state'] = base.JOB_STATE_DONE
+                trial['result'] = result
+                trial['refresh_time'] = coarse_utcnow()
+        self.trials.refresh()
+
+    def run(self, num_trials):
+        """
+        :param num_trials: 
+        :return: 
+        """
+        trials = self.trials
+        algo = self.algo
+
+        for j in range(num_trials):
+            new_ids = trials.new_trial_ids(1)
+            self.trials.refresh()
+            new_trials = algo(new_ids, self.domain, trials,
+                              self.rstate.randint(2 ** 31 - 1))
+            self.trials.insert_trial_docs(new_trials)
+            self.trials.refresh()
+
+        self.serial_evaluate()
+
+
+def dataframe_to_trials(data: DataFrame):
+    trials = base.Trials()
+    names = list(data)[:-1]
+    for index, row in data.iterrows():
+        misc = dict(tid=index, cmd=('domain_attachment', 'FMinIter_Domain'), workdir=None,
```

Comment:
What is `misc`?

Reply:
`misc` contains the coordinates of the next suggestion (i.e. what we really want). it's weird that such an important thing is contained inside a dictionary called `misc` but it's the way the framework work and there's really no other way to change it without changing the framework significantly.

Reply:
Yeah, we don't want to change the internals of the framework at this point. It would be useful to provide comments for not-so-clear variables and data structures.